src/python/WMCore/WMSpec/WMTask.py

```diff
@@ -537,22 +537,9 @@ def setJobResourceInformation(self, timePerEvent=None, sizePerEvent=None, memory
             performanceParams.timePerEvent = timePerEvent or getattr(performanceParams, "timePerEvent")
         if sizePerEvent or getattr(performanceParams, "sizePerEvent", None):
             performanceParams.sizePerEvent = sizePerEvent or getattr(performanceParams, "sizePerEvent")
-
-        # special handling for memory overwrite during assignment
-        if isinstance(memoryReq, dict):
-            taskMemory = memoryReq.get(self.name())
-        else:
-            taskMemory = memoryReq
-
-        if self.taskType() in ["Merge", "Harvesting", "Cleanup", "LogCollect"]:
-            # we don't want to touch memory for these task types
-            pass
-        elif taskMemory or getattr(performanceParams, "memoryRequirement", None):
+        if memoryReq or getattr(performanceParams, "memoryRequirement", None):
```

Comment:
That's right. We skip those task types and they get some default values in JobCreator (or submitter plugin).